node/core/backing/src/lib.rs

```diff
@@ -1357,6 +1423,18 @@ async fn handle_validated_candidate_command<Context>(
 									seconded_at_depth.insert(depth, candidate_hash);
 								}
 							}
+							if !is_max_depth {
+								// send notifications for unblocked candidates.
+								let para = candidate.descriptor.para_id;
+								let head_data_hash = candidate.descriptor.para_head;
+								if let Some(requests) =
+									state.seconding_check_requests.remove(&(para, head_data_hash))
+								{
+									for req in requests {
+										let _ = req.tx.send(true);
```

Comment:
Logic here makes sense, but it seems we should have a similar check/loop upon new active leaves, as candidates converge to depth 0 or removal from new fragment trees.

Reply:
On second glance this looks incorrect, or at least not totally correct.

The main things the Seconded Sanity Check is concerned about are:
  1. Are we aware of any place the candidate can fit in a known fragment tree?
  2. If so, have we seconded anything conflicting with that depth?

That is, we need (1) to be true and (2) to be false. (1) can change from false to true either on seeing new candidates `Seconded` by ourselves _or_ by others. (1) can also change when we get new active leaves and space opens up in a fragment tree - as I mentioned in the above comment. It makes more sense for this logic to run after statement import (i.e. in the same place we send `ProspectiveParachainsMessage::CandidateSeconded`) and on new active leaves.

Reply:
Well, re #6002 we would like to accept collations built upon either `Backed` candidates or those which are `Seconded` and which we've validated locally (which are not necessarily the same thing as `Seconded` locally).

Capturing the nuance between these two positions is probably more code than it's worth at the moment. To keep things simple it'd be best to just run this logic after a candidate is backed (and also on new relay-parents).

Reply:
1. > Well, re https://github.com/paritytech/polkadot/issues/6002 we would like to accept collations built upon either Backed candidates or those which are Seconded and which we've validated locally (which are not necessarily the same thing as Seconded locally).

Isn't accepting collations built on top of something we've validated locally the best option? Solves #6002 right away if I'm not missing some case.

2. As for the activated leaves, yes, missed it.